server/core/connector.go

```diff
@@ -397,7 +397,7 @@ func (conn *BridgeConnector) subscribeToJetStream(subject string, queueName stri
 			conn.bridge.Logger().Tracef("%s received message", conn.String())
 		}
 
-		key := conn.calculateKey(conn.config.Subject, conn.config.DurableName)
+		key := conn.calculateKey(msg.Subject, conn.config.DurableName)
```

Comment:
Maybe we could add some tests to prevent regression and clearly show intentions and use cases for this change.

Reply:
test added